lib/domgen/sync/templates/abstract_sync_temp_population_impl.java.erb

```diff
@@ -133,26 +134,26 @@ public abstract class <%= data_module.sync.abstract_sync_temp_population_impl_na
-          " WHERE O.<%= entity.sync.master_entity.attribute_by_name(:MappingID).sql.quoted_column_name %> IS NOT NULL AND S.<%= entity.sync.master_entity.attribute_by_name(:MappingSource).sql.quoted_column_name %> = ?";
+          "   #{ entity.sync.sync_temp_entity.sql.qualified_table_name } AS S " +
+          " LEFT JOIN #{ entity.sync.sync_temp_entity.sql.qualified_table_name } AS O ON O.#{ entity.sync.sync_temp_entity.attribute_by_name(:MappingID).sql.quoted_column_name } = S.#{ entity.sync.sync_temp_entity.attribute_by_name(:MappingID).sql.quoted_column_name } AND O.#{ entity.sync.sync_temp_entity.attribute_by_name(:MappingSource).sql.quoted_column_name } = S.#{ entity.sync.sync_temp_entity.attribute_by_name(:MappingSource).sql.quoted_column_name } AND O.#{ entity.sync.sync_temp_entity.attribute_by_name(:SyncTempID).sql.quoted_column_name } <> S.#{ entity.sync.sync_temp_entity.attribute_by_name(:SyncTempID).sql.quoted_column_name } " +
+          " WHERE O.#{ entity.sync.master_entity.attribute_by_name(:MappingID).sql.quoted_column_name } IS NOT NULL AND S.#{ entity.sync.master_entity.attribute_by_name(:MappingSource).sql.quoted_column_name } = ?").gsub('"','\"')%>";
         if ( !verify<%= entity.qualified_name.gsub('.','') %>ViaSql( mappingSourceCode, sql, "MappingID Uniqueness", recorder ) )
         {
           validationFailed = true;
         }
       }
 <% entity.sql.indexes.select{|index| index.unique? && !index.attribute_names.any?{|k| %w(CreatedAt DeletedAt).include?(k.to_s)}}.each do |index|
-   column_names = index.attribute_names.collect {|attr_name| a = entity.attribute_by_name(attr_name); entity.sync.sync_temp_entity.attribute_by_name(a.reference? ? "#{a.name}MappingID" : a.name).sql.quoted_column_name }
+   column_names = index.attribute_names.collect {|attr_name| a = entity.attribute_by_name(attr_name); entity.sync.sync_temp_entity.attribute_by_name(a.reference? ? a.name.to_s + 'MappingID' : a.name).sql.quoted_column_name }
 -%>
       {
         final String sql =
-          "SELECT S.* " +
+     "<%=("SELECT S.* " +
           "FROM " +
-          "  <%= entity.sync.sync_temp_entity.sql.qualified_table_name %> AS S " +
-          "JOIN ( SELECT <%= column_names.join(',') %> FROM <%= entity.sync.sync_temp_entity.sql.qualified_table_name %> GROUP BY <%= column_names.join(',') %> HAVING COUNT(*) > 1 ) J ON <%= column_names.collect{|c| "J.#{c} = S.#{c}"}.join(' AND ') %> " +
-          "WHERE S.<%= entity.sync.master_entity.attribute_by_name(:MappingSource).sql.quoted_column_name %> = ?";
+          "  #{ entity.sync.sync_temp_entity.sql.qualified_table_name } AS S " +
+          "JOIN ( SELECT #{ column_names.join(',') } FROM #{ entity.sync.sync_temp_entity.sql.qualified_table_name } GROUP BY #{ column_names.join(',') } HAVING COUNT(*) > 1 ) J ON #{ column_names.collect{|c| 'J.' + c + ' = S.' + c}.join(' AND ') } " +
```

Comment:
Any reason why you keep replacing interpolation with concaternation? Seemingly interpolation is easier